client/webserver/http.go

```diff
@@ -190,6 +195,126 @@ func (s *WebServer) handleOrders(w http.ResponseWriter, r *http.Request) {
+	wf, ok := w.(http.Flusher)
+	if !ok {
+		log.Errorf("unable to flush streamed data")
+		http.Error(w, "unable to flush streamed data", http.StatusBadRequest)
+		return
+	}
+
+	filter := new(core.OrderFilter)
+	err := r.ParseForm()
+	if err != nil {
+		log.Errorf("error parsing form for export order: %v", err)
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return
+	}
+
+	filter.Hosts = r.Form["hosts"]
+	assets := r.Form["assets"]
+	filter.Assets = make([]uint32, len(assets))
+	for k, assetStrID := range assets {
+		assetNumID, err := strconv.ParseUint(assetStrID, 10, 32)
+		if err != nil {
+			log.Errorf("error parsing asset id: %v", err)
+			http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+			return
+		}
+		filter.Assets[k] = uint32(assetNumID)
+	}
+	statuses := r.Form["statuses"]
+	filter.Statuses = make([]order.OrderStatus, len(statuses))
+	for k, statusStrID := range statuses {
+		statusNumID, err := strconv.ParseUint(statusStrID, 10, 16)
+		if err != nil {
+			http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+			log.Errorf("error parsing status id: %v", err)
+			return
+		}
+		filter.Statuses[k] = order.OrderStatus(statusNumID)
+	}
+
+	ords, err := s.core.Orders(filter)
+	if err != nil {
+		log.Errorf("error retrieving order: %v", err)
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Disposition", "attachment; filename=orders.csv")
+	w.Header().Set("Content-Type", "text/csv")
+	w.WriteHeader(http.StatusOK)
+	csvWriter := csv.NewWriter(w)
+	csvWriter.UseCRLF = strings.Contains(r.UserAgent(), "Windows")
+
+	err = csvWriter.Write([]string{
+		"Host",
+		"Base",
+		"Quote",
+		"Ask",
+		"Offer",
+		"Type",
+		"Side",
+		"Time in Force",
+		"Status",
+		"Rate",
+		"Filled (%)",
+		"Settled (%)",
+		"Quantity",
+		"Time",
+	})
+	if err != nil {
+		log.Errorf("error writing CSV: %v", err)
+		return
+	}
+	csvWriter.Flush()
+	err = csvWriter.Error()
+	if err != nil {
+		log.Errorf("error writing CSV: %v", err)
+		return
+	}
+	wf.Flush()
+
+	for _, ord := range ords {
+		ordReader := orderReader{ord}
+		timestamp, err := encode.UnixTimeMilliLocal(int64(ord.Stamp)).MarshalText()
+		if err != nil {
+			log.Errorf("error writing CSV: %v", err)
+			return
+		}
+
+		err = csvWriter.Write([]string{
+			ord.Host,                     // Host
+			ord.BaseSymbol,               // Base
+			ord.QuoteSymbol,              // Quote
+			ordReader.AskString(),        // Ask
+			ordReader.OfferString(),      // Offer
+			ordReader.Type.String(),      // Type
+			ordReader.SideString(),       // Side
+			ord.TimeInForce.String(),     // Time in Force
+			ordReader.StatusString(),     // Status
+			ordReader.SimpleRateString(), // Rate
+			ordReader.FilledPercent(),    // Filled
+			ordReader.SettledPercent(),   // Settled
+			ordReader.OfferString(),      // Quantity
```

Comment:
"Offer" and "Quantity" are both set to `ordReader.OfferString()`. Was "Quantity" supposed to be effectively `precision8(ord.Qty)`?

Reply:
I just copied this from the previous PR, but I've made some changes to the columns now. Let me know what you think.

Reply:
Thanks, I don't know what was going on with the previous columns.  I like this average rate you added, which makes sense when there are multiple matches that aren't necessarily same rate.  Really nice.